packages/toolpad-app/src/ThemeContext.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { ThemeProvider as MuiThemeProvider, createTheme } from '@mui/material/styles';
+import { deepmerge } from '@mui/utils';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/material/utils';
+import { getDesignTokens, getThemedComponents, getMetaThemeColor } from './theme';
+
+interface ThemeProviderProps {
+  children?: React.ReactNode;
+}
+
+export type PaletteModeOptions = 'light' | 'dark';
+
+export const DispatchContext = React.createContext<
```

Comment:
You may want to use `createProvidedContext`?

Reply:
I imagine that `createProvidedContext` and `useToolpadThemeMode` won't both be required together, since:

```ts
export const [useThemeModeDispatch, ThemeModeDispatchContextProvider] = createProvidedContext<React.Dispatch<React.SetStateAction<PaletteModeOptions>>>(
    'ThemeModeDispatch',
  );
 ```
 
 creates a `useThemeModeDispatch` which isn't needed?

Reply:
Yeah, if we use `useLocalStorageState`, I think we may not need this context at all?